<?xml version="1.0"?>

<!--
Mon Oct 10 19:20:36 EEST 2016, Nikos Koukis

Launchfile  responsible for the mrpt_graphslam_2d application in a multi-robot
slam agent setup. It practically provides a wrapper around the graphslam.launch
file.

Refer to the README.md file for usage information.
DO NOT USE.
-->

<launch>
	<!-- LAUNCH ARGUMENTS -->
	<arg name="output" default="screen"/>
	<arg name="config_file" default="$(find mrpt_graphslam_2d)/config/ros_odometry_2DRangeScans.ini"  />
	<arg name="do_publish_gridmap_topic" default="true" />

	<arg name="anchor_frame_ID" default="map" />
	<arg name="base_link_frame_ID" default="base_link"/>
	<arg name="odometry_frame_ID" default="odometry"/>
	<arg name="laser_frame_ID" default="laser"/>

	<arg name="NRD" default="CFixedIntervalsNRD" />
	<arg name="ERD" default="CLoopCloserERD_MR" />
	<arg name="GSO" default="CLevMarqGSO" />

  <!-- Should we also start an instance of rviz ? -->
  <arg name="start_rviz" default="false" />
  <arg name="rviz_file" default="$(find mrpt_graphslam_2d)/rviz/mr_graphslam.rviz" />

	<arg name="disable_MRPT_visuals" default="false" />

	<include file="$(find mrpt_graphslam_2d)/launch/graphslam.launch"
	  pass_all_args="true">
  </include>

  <!-- Robot Geometrical Configuration -->
  <!--<node pkg="tf2_ros" type="static_transform_publisher" name="base_link_laser_broadcaster" args="0 0 0.05 0 0 0 1 $(arg base_link_frame_ID) $(arg laser_frame_ID)" output="$(arg output)" />-->
</launch>
